src/App.tsx

```diff
@@ -1,223 +1,232 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+  FormEvent,
+} from 'react';
+import classnames from 'classnames';
+import {
+  createTodo,
+  getTodos,
+  removeTodo,
+  updateTodo,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Footer } from './components/Footer';
+import { TodosList } from './components/TodosList';
+import { FilterType } from './types/FilterStatus';
+import { Todo } from './types/Todo';
+import { Error } from './types/Error';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const inputField = useRef<HTMLInputElement>(null);
+  const [error, setError] = useState<Error | null>(null);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [title, setTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [selectedTodos, setSelectedTodos] = useState<number[]>([]);
+  const [toggle, setToggle] = useState(true);
+  const [selectedTodo, setSelectedTodo] = useState(0);
+
+  if (error) {
+    setTimeout(() => {
+      setError(null);
+      setIsAdding(false);
+    }, 3000);
+  }
+
+  let userId = 0;
+
+  if (user?.id) {
+    userId = user?.id;
+  }
+
+  const filteredTodos = todos.filter(todo => {
+    switch (filterType) {
+      case FilterType.All:
+        return todo;
+
+      case FilterType.Active:
+        return !todo.completed;
+
+      case FilterType.Completed:
+        return todo.completed;
+
+      default:
+        return null;
+    }
+  });
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
-    }
+    inputField.current?.focus();
+
+    getTodos(userId)
+      .then(setTodos)
+      .catch(() => setError(Error.LOADING));
```

Comment:
Let’s use async await. It gives you a simpler way to work with asynchronous promise-based code